app/services/DynamoChannelTestsAudit.scala

```diff
@@ -0,0 +1,93 @@
+package services
+
+import com.typesafe.scalalogging.StrictLogging
+import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}
+import io.circe.syntax.EncoderOps
+import io.circe.{Decoder, Encoder}
+import models.ChannelTest
+import software.amazon.awssdk.services.dynamodb.DynamoDbClient
+import software.amazon.awssdk.services.dynamodb.model.{AttributeValue, _}
+import zio.{ZEnv, ZIO}
+import models.DynamoErrors._
+import DynamoChannelTestsAudit.{ChannelTestAudit, getTimeToLive}
+import utils.Circe.{dynamoMapToJson, jsonToDynamo}
+import zio.blocking.effectBlocking
+
+import java.time.OffsetDateTime
+import scala.jdk.CollectionConverters.{ListHasAsScala, MapHasAsJava}
+
+object DynamoChannelTestsAudit {
+  // The model that we write to the audit table
+  case class ChannelTestAudit[T : Encoder : Decoder](
```

Comment:
We can create an audit entry for any item of type `T` which has a JSON encoder + decoder